### PR TITLE
fix versions for determinist sensitive crates

### PR DESCRIPTION
**Pull request by modship**
* [x] document all added functions
* [ ] try in sandbox /simulation/labnet
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification

part of #3899  

https://github.com/massalabs/massa/issues/3899#issuecomment-1533212663

Determinism-sensitive crates:
* [X] rand in massa-pos-worker
* [X] rand_distr in massa-pos-worker
* [X] rand_xoshiro in massa-pos-worker
* [X] blake3 in massa-hash
* [X] bs58 in massa-hash 
* [X] ed25519-dalek in massa-signature
* [X]  bs58 in massa-signature
* [X] `nom` everywhere
* [ ] `rand` in massa-signature ?
* [ ] the future state merkle tree crate (lsmtree = "0.1.1" in massa-db) of https://github.com/massalabs/massa/pull/3783/files#diff-86b01d6802887cf1e2848e1375148362ac3897932bd7f13ffca986ab5f14f952

### CODE REVIEW

**Comment by modship** [comment]
https://github.com/massalabs/massa/issues/3899#issuecomment-1564448781

File: massa-models/Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ directories = "4.0"
 config = "0.13"
 bs58 = { version = "0.4", features = ["check"] }
```
Comment (damip) [inline massa-models/Cargo.toml:18]: define bs58 version

File: massa-models/Cargo.toml [hunk 2]
```diff
@@ -17,7 +17,7 @@ directories = "4.0"
...
 bitvec = { version = "1.0", features = ["serde"] }
```
Comment (damip) [inline massa-models/Cargo.toml:19]: define bitvec version (seed determinism)

File: massa-models/Cargo.toml [hunk 3]
```diff
@@ -17,7 +17,7 @@ directories = "4.0"
...
-nom = "7.1"
+nom = "=7.1"
 
 # custom modules
 massa_hash = { path = "../massa-hash" }
```
Comment (damip) [inline massa-models/Cargo.toml:23]: dependency `transition = { git = "https://github.com/massalabs/transition.git" }` => depend on a specific tag

File: massa-protocol-exports/Cargo.toml [hunk 4]
```diff
@@ -7,7 +7,7 @@ edition = "2021"
 [dependencies]
 displaydoc = "0.2"
 thiserror = "1.0"
-nom = "7.1"
+nom = "=7.1"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 peernet = { git = "https://github.com/massalabs/PeerNet", rev = "0ad0e2e" }
```
Comment (damip) [inline massa-protocol-exports/Cargo.toml:13]: TODO: tag

File: massa-protocol-worker/Cargo.toml [hunk 5]
```diff
@@ -10,7 +10,7 @@ rand = "0.8"
 parking_lot = "0.12"
 crossbeam = "0.8"
 serde_json = "1.0"
-nom = "7.1"
+nom = "=7.1"
 num_enum = "0.5"
 peernet = { git = "https://github.com/massalabs/PeerNet", rev = "0ad0e2e" }
```
Comment (damip) [inline massa-protocol-worker/Cargo.toml:15]: TODO tag

File: massa-signature/Cargo.toml [hunk 6]
```diff
@@ -7,12 +7,12 @@ edition = "2021"
...
-bs58 = { version = "0.4", features = ["check"] }
+bs58 = { version = "=0.4", features = ["check"] }
 displaydoc = "0.2"
-ed25519-dalek = { version = "1.0", features = ["batch"] }
+ed25519-dalek = { version = "=1.0", features = ["batch"] }
 serde = { version = "1.0", features = ["derive"] }
 thiserror = "1.0"
-nom = "7.1"
+nom = "=7.1"
 rand = "0.7"
```
Comment (damip) [inline massa-signature/Cargo.toml:16]: dep `transition = { git = "https://github.com/massalabs/transition.git" }` => depend on fixed tag

File: massa-versioning/Cargo.toml [hunk 7]
```diff
@@ -10,7 +10,7 @@ machine = { git = "https://github.com/antifuchs/machine", branch = "fix-workspac
 parking_lot = "0.12"
 thiserror = "1.0"
 num_enum = "0.5"
-nom = "7.1"
+nom = "=7.1"
 tracing = "0.1"
 
 # custom module
```
Comment (damip) [inline massa-versioning/Cargo.toml:16]: TODO fixate chrono version

**Comment by damip** [comment]
* yes, the version of the future merkle tree needs to be fixed
* would be nice to fixate rand as well in general (but not 100% necessary)

**Review by damip — APPROVED** [review]

**Pull request merged**